### DISCUSSION TITLE
Puma installation question, Windows, signals

### DISCUSSION THREAD

**Discussion by AhzDeveloperlol** [puma/puma#3050] [Q&A]
Whenever I run the ```puma``` command this error shows up and the website wont load
here is  the error:

```
*** SIGUSR2 not implemented, signal based restart unavailable!
*** SIGUSR1 not implemented, signal based restart unavailable!      
*** SIGHUP not implemented, signal based logs reopening unavailable!
```
Please help me fix this issue

**Comment by MSP-Greg** [comment]
From your project folder, try:
```ps1
# below installs gems in project folder
bundle config set --local path ./vendor/bundle
bundle install --jobs 4 --retry 3
bundle exec puma
````

**Reply by MSP-Greg** [reply]
Regarding the log you've shown, supported signals on a platform/OS can be shown with `ruby -e "puts Signal.list"`.  Windows supports very few signals, and some that it shows as 'supported' don't work very well.

So, the log you've shown isn't errors, it's showing that some signals aren't supported by Windows.  One can send several commands to Puma once it's started.  They can be sent by a http request or by sending a signal to Puma's `Process.pid`, which is shown on the following line of the startup log::
```
*          PID: <pid number>
```

**Comment by MSP-Greg** [comment]
@AhzDeveloperlol

I checked again this morning, and something is wrong, possibly your Windows install, not sure.  This should run with a simple `puma` on the command line.  So, from your project folder, type the following:
```
bundle install
```
Should show:
```
Fetching gem metadata from https://rubygems.org/..
Using bundler 2.3.26
Fetching nio4r 2.5.8
Fetching rack 3.0.3
Installing rack 3.0.3
Installing nio4r 2.5.8 with native extensions
Fetching puma 6.0.1
Installing puma 6.0.1 with native extensions
Bundle complete! 2 Gemfile dependencies, 4 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
```

Then, typing:
```
puma
```

Should show:
```
*** SIGUSR2 not implemented, signal based restart unavailable!
*** SIGUSR1 not implemented, signal based restart unavailable!
*** SIGHUP not implemented, signal based logs reopening unavailable!
Puma starting in single mode...
* Puma version: 6.0.1 (ruby 3.1.3-p185) ("Sunflower")
*  Min threads: 0
*  Max threads: 5
*  Environment: development
*          PID: 12248
* Listening on http://0.0.0.0:9292
```

I just saw that you commited files from the first script I posted, I'll try to see what's wrong with them.

**Comment by MSP-Greg** [comment]
@AhzDeveloperlol

I'm fairly certain your problem is with 'Windows Firewall'.  The code you added to https://github.com/AhzDeveloperlol/website-made-in-ruby works.

**Reply by AhzDeveloperlol** [reply]
Hey, sorry for not replying for a while but I have had issues with windows firewall before so I am gonna try it. I will tell you the results
